### DISCUSSION TITLE
Unable to connect to my milvus db instance deployed on EKS cluster

### DISCUSSION THREAD

**Discussion by AbhiMewt** [milvus-io/milvus#23728] [Q&A and General discussion]
Hi ,
I have deployed a milvus on AWS EKS cluster and trying to connect to it on grpc port 
but everytime it pops an error as connection timedout error 

trying to connect using :
```

from pymilvus import connections, FieldSchema, CollectionSchema, DataType, Collection, utility
connections.connect(host=MILVUS_HOST, port=MILVUS_PORT)
```

i am using loadbalancer ip as host ip and the load balancer port shown is like 19530:32670 -> using 32670
```

    115                 return
    116             except grpc.FutureTimeoutError as e:
--> 117                 raise MilvusException(Status.CONNECT_FAILED,
    118                                       f'Fail connecting to server on {self._address}. Timeout') from e
    119 

MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 192.168.12.27:19530. Timeout)>
```

**Comment by yhmo** [comment]
The returned error is "Fail connecting to server on 192.168.12.27:19530", did you change the MILVUS_PORT to 32670?

**Reply by AbhiMewt** [reply]
yes still the error is same

**Reply by yhmo** [reply]
`connections.connect(host='192.168.12.27', port="32670")`, is this call return "Fail connecting to server on 192.168.12.27:19530"?

**Reply by TheRhicrosoradile1** [reply]
Yes

**Reply by LoveEachDay** [reply]
Could you provide the detailed information about the loadbalancer? Did you use the `NodePort` servicetype of `LoadBalancer` type for milvus proxy service?

**Comment by ramialt** [comment]
Edit: Figured out the issue, due to own error

Hi, I also have this issue. I followed the [AWS EKS guide](https://milvus.io/docs/eks.md) and got all pods running and working. When port-forwarding to the load balancer (`kubectl port-forward svc/<milvus-svc> 19530`) I can connect to localhost:19530 just fine. But external connection does not work. 

When running `kubectl get svc`, I see LoadBalancer type listed along with external IP. I later changed [this value in helm values.yaml](https://github.com/milvus-io/milvus-helm/blob/master/charts/milvus/values.yaml#L59) to LoadBalancer afterwards and re-deployed as well, but that didn't seem to change anyway.

My connect script looks like `connections.connect('default', host='<id>.<aws region>.[elb.amazonaws.com](http://elb.amazonaws.com/)', port='19530')` and I get error:
```
pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on <id>.<aws region>.elb.amazonaws.com:19530. Timeout)>
```

~~Maybe there is a missing step in AWS EKS guide to enable external LoadBalancer?~~

### Edit
Nevermind, this was a personal network setup issue. Just had to change the port the load balancer was listening on.

**Reply by jmugan** [reply]
I'm having this exact issue? What was the solution? Change the port the load balancer was listening on? How do you do that? Isn't that 19530?

**Reply by ramialt** [reply]
Yep, I had to change the port the load balancer was listening on, otherwise my personal network would ignore the connection attempt. 
You can change this port number on this line of the [helm values config file](https://github.com/milvus-io/milvus-helm/blob/master/charts/milvus/values.yaml#L60). As you can see, the default is 19530, but that can be changed to probably any port number.

**Reply by mahaveer08** [reply]
As I am trying to deploy using latest documentation but failed to up all the pods as I am deploying on existing cluster so should I need to configure anything?

**Comment by jmugan** [comment]
Sorry to be dense, but I'm still getting this error both on 2.2 and 2.3 instructions at https://milvus.io/docs/eks.md. It's listening on 19530 and I'm trying to connect on 19530.

`kubectl get svc -n milvus` gives
`milvus-demo                    LoadBalancer   10.100.104.92    milvus-service-c5e162d2cbb28e6d.elb.us-east-1.amazonaws.com   19530:30643/TCP,9091:32694/TCP        5m27s` 

and I'm trying to connect with
`connections.connect("default", host="milvus-service-c5e162d2cbb28e6d.elb.us-east-1.amazonaws.com", port="19530")`

I set `service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing`

I can't seem to ping it either. Any suggestions?

**Reply by ramialt** [reply]
Looks like they changed the AWS install guide quite a bit from when I last messed with this (which is good), so not sure how much I can help there.

My thoughts are that either you aren’t able to connect to your load balancer, or your load balancer isn’t able to connect to your Milvus cluster.

For the former option, have you checked if it could be your firewall? 19530 isn’t exactly a standard port. Try deploying Milvus to use something like port 443, by changing the value in the `milvus.yaml` config file. If it was just your firewall, you should be able to connect to the load balancer after that.

For the latter option, you would have to connect to your LoadBalancer EC2 instances and see if you can hit any Milvus nodes. I’m not too experienced with this, so I can’t really advise you more than that. But if the first option didn’t work, maybe someone else here can jump in.

Best of luck

**Reply by jmugan** [reply]
I'll try that, thanks for your help!

**Comment by Anshuman-gitt** [comment]
i have followed only line by steps of https://milvus.io/docs/azure.md i have deployed on azure but when i ping on the external ip it is showing 100% packet loss and when i tried to connect it with externally it is showing an error of (i have used loadbalancer)
pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 4.246.41.153:27017. Timeout> Disconnected from Milvus

**Reply by LoveEachDay** [reply]
@Anshuman-gitt You cannot ping the service ip with load balancer type. Try check the port connection using `nc` or `telnet`.
You'd also use the `19530` port instead if you are following the above attached documentation.

**Reply by Anshuman-gitt** [reply]
i have tried 19530 nd 9091 same error is showing
